### ISSUE TITLE
Is "const" in snippets causing confusion?  Should we have done only "let"-s?

### ISSUE THREAD

**Issue by Zlatkovsky** [OfficeDev/office-js-snippets#29]
@jakobpn , would be curious to hear your opinion on this.  Internally and externally, I feel like folks have had a hard time reasoning about what a `const sheet =` is (e.g., thinking that it's a protected sheet that you can't add to, etc.).

Should we just bulk-change all `const`-s to `let`-s?

**Comment by jakobpn** [comment]
I think we should take advantage of "const" providing a safety net that help ensure that if you don't change a reference to an object by mistake. My vote is to keep the snippets as is.

**Comment by Zlatkovsky** [comment]
Agreed in principle.  In practice, though, I have yet to see it catch an error for me.  Whereas I have seen it cause confusion...

I'm split on this (for my own code, I would use `const` -- but I'm actually not sure that I would when explaining in, say, my book).  So as you interact with customers, would be curious to hear what their impressions are.

**Comment by jakobpn** [comment]
So far it I don't think I have observed confusion, but it is something that I will look after.

**Comment by Rick-Kirkham** [comment]
FWIW, I find it puzzling. In every other language I know, `const` is used for something whose value is known and assigned by the developer as a literal at coding time.  It is weird to see a `const `initialized at runtime with the return value of a method.

**Comment by AlexJerabek** [comment]
@Zlatkovsky and @Rick-Kirkham , I'm curious if you've had further thoughts on this matter a year and a half later. My thoughts are that tslint enforces const where possible, so we should try to conform to what's ostensibly our own (MSFT) guidance.

**Comment by Rick-Kirkham** [comment]
@AlexJerabek @Zlatkovsky I've reversed myself. I now think we should use const as it is intended in TypeScript. It is initially confusing to people who know other languages with "const" because it has a different meaning in TS, but I think now that that's just part of the process of learning a new language.
